domain/application/state/state.go

```diff
@@ -1328,8 +1328,36 @@ WHERE uuid = $applicationID.uuid;
 		return applicationerrors.ApplicationIsDead
 	}
 
-	if err := result.ID.Validate(); err != nil {
-		return internalerrors.Errorf("invalid application ID %q: %w", result.ID, err)
+	return nil
+}
+
+// checkApplicationWithNameExists checks if the application exists and is not dead. It's
+// possible to access alive and dying applications, but not dead ones. If the
+// application is dead, [applicationerrors.ApplicationIsDead] is returned. If
+// the application is not found, [applicationerrors.ApplicationNotFound] is
+// returned.
+func (st *State) checkApplicationWithNameExists(ctx context.Context, tx *sqlair.TX, ident applicationName) error {
+	query := `
+SELECT &applicationLife.*
+FROM application
+WHERE name = $applicationName.name;
+`
+	stmt, err := st.Prepare(query, ident, applicationLife{})
+	if err != nil {
+		return internalerrors.Errorf("preparing query for application %q: %w", ident.Name, err)
+	}
+
+	var result applicationLife
+	err = tx.Query(ctx, stmt, ident).Get(&result)
+	if errors.Is(err, sql.ErrNoRows) {
+		return applicationerrors.ApplicationNotFound
+	} else if err != nil {
+		return internalerrors.Errorf("checking application %q exists: %w", ident.Name, err)
+	}
+
+	// If we're not alive or dying, then the application is dead.
+	if result.LifeID > domainlife.Dying {
+		return applicationerrors.ApplicationIsDead
 	}
```

Comment:
If someone messes up the IDs, this test will be broken. Swap it for a switch statement.

```suggestion
	switch result.LifeID  {
	case domainlife.Dead:
		return applicationerrors.ApplicationIsDead
	default:
	    return nil
	}
```

Reply:
I will change this both times